internal/webhooks/cluster_test.go

```diff
@@ -1266,18 +1403,28 @@ func TestClusterTopologyValidation(t *testing.T) {
 			// Sets up the fakeClient for the test case.
 			fakeClient := fake.NewClientBuilder().
 				WithObjects(class).
+				WithObjects(tt.additionalObjects...).
 				WithScheme(fakeScheme).
 				Build()
 
+			if tt.clusterCacheTrackerReader == nil {
+				tt.clusterCacheTrackerReader = &fakeClusterCacheTracker{client: fake.NewFakeClient()}
```

Comment:
Looks like we never set tt.clusterCacheTrackerReader. Do we anticipate that we'll need it later? If not I would drop clusterCacheTrackerReader from the table struct